pkg/worker/worker.go

```diff
@@ -36,6 +36,7 @@ import (
 func GatherResults(waitfile string, url string) error {
 	var inputFileName []byte
 	var err error
+	var outfile *os.File
 
 	// just loop looking for a file.
 	done := false
```

Comment:
I know this is out of scope, but what about

for {
 // do work
 if err == nil {
     // success!!
     break
 }
 // log and wait
}

Reply:
Patches welcome ;-)  
fwiw I didn't write this part of the code, I'm just trying to throw some fixes in before release.